### ISSUE TITLE
metadata reliability

### ISSUE THREAD

**Issue by guestisp** [lizardfs/lizardfs#701]
Some questions about metadata reliability.

AFAIK, all metadata are kept in RAM, for performance, on master server.
Then we have one ore more shadow servers, that are synced in realtime with the master (something like MySQL master-slave replication).

So, in case of master failure, a shadow should be able to became a new master with no data loss.
By using multiple shadows, we should be able to reduce the risk of data loss (we have to loose the master and multiple shadows at the same time)

Other than this, master will hourly dump the whole metadata archive. This archive could be restored in case of disaster (obviously, we loose everything that was changed between the last dump and the crash)

Then , we also have metaloggers, that will fetch metadata from master server once per day.

To summarize, data loss (and with "data loss" I also mean a restore with an outdated dump) could happen if and only if: all master and shadows are lost.

If we have to restore from metalogger, data loss should be assured, because metalogger are not updated in realtime. (exactly like restoring a database from an `.sql` dump). The same should be when restoring from master automatic dump, made once per hours, right ?

**Comment by 4Dolio** [comment]
Metalogger do record changelogs. Just like masters and shadows..

**Comment by guestisp** [comment]
> Metalogger do record changelogs. Just like masters and shadows..

In realtime ? I've read that a dump is made every 24 hours (as default). And at which frequency a changelog is made ?

**Comment by 4Dolio** [comment]
changelogs are in real time, and human readable.  Find your newest one and tail it some time..

**Comment by guestisp** [comment]
> changelogs are in real time, and human readable. Find your newest one and tail it some time..

my test cluster is currently down, i don't have a working lizard for some weeks more

**Comment by guestisp** [comment]
> And the default for full bin meta dumps is 1 hour i think

Seems to be 24 hours as default: 
https://github.com/lizardfs/lizardfs/blob/master/doc/mfsmetalogger.cfg.5.txt#L48

**Comment by 4Dolio** [comment]
Shrugs. I stand corrected. Anyway..
